File: internal/content/testdata/bench/typescript/all-permutations-of-array.ts
function all_permutations(nums: number[]): number[][] {
    if (nums.length === 0) return [[]];
    const result: number[][] = [];
    for (let i = 0; i < nums.length; i++) {
        const n = nums[i];
        const rest = nums.slice(0, i).concat(nums.slice(i + 1));
        for (const perm of all_permutations(rest)) {
            result.push([n, ...perm]);
        }
    }
    return result;
}

File: internal/content/testdata/bench/go/all-permutations-of-array.go
func all_permutations(nums []int) [][]int {
	if len(nums) == 0 {
		return [][]int{{}}
	}
	result := [][]int{}
	for i, n := range nums {
		rest := make([]int, 0, len(nums)-1)
		rest = append(rest, nums[:i]...)
		rest = append(rest, nums[i+1:]...)
		for _, perm := range all_permutations(rest) {
			combo := make([]int, 0, len(perm)+1)
			combo = append(combo, n)
			combo = append(combo, perm...)
			result = append(result, combo)
		}
	}
	return result
}
